### PR TITLE
review-likes-search-page: add like-dislike actions

### PR DESCRIPTION
**Pull request by ivanshamis**

### CODE REVIEW

File: frontend/src/components/review/PlacePreviewReviewItem.vue [hunk 1]
```diff
@@ -37,12 +40,31 @@ export default {
...
+            required: true
         }
     },
     computed:{
         userName() {
             return this.review.user.first_name + ' ' + this.review.user.last_name;
         }
+    },
+    methods: {
+        ...mapActions('place', ['likePlaceReview', 'dislikePlaceReview']),
```
Comment (nemoipaha) [inline frontend/src/components/review/PlacePreviewReviewItem.vue]: move likePlaceReview', 'dislikePlaceReview' methods inside review module

File: frontend/src/store/modules/place/actions.js [hunk 2]
```diff
@@ -143,6 +143,96 @@ export default {
...
     },
 
+    likePlaceReview: (context, data) => {
+        httpService.post(`reviews/${data.id}/like`)
+            .then( (res) => {
+                let review = context.state.places.find( (place) => { 
+                    return place.id === parseInt(data.placeId); 
+                }).review;
+                if (review.like === STATUS_NONE) {
+                    context.commit('SET_PLACE_REVIEW_LIKE_STATE', {
```
Comment (nemoipaha) [inline frontend/src/store/modules/place/actions.js]: context.commit('SET_PLACE_REVIEW_LIKE_STATE', {
            reviewId: reviewId,
            likeState: STATUS_LIKED
});

change `placeId: data.placeId`, to reviewId: data.id

File: frontend/src/store/modules/place/actions.js [hunk 3]
```diff
@@ -143,6 +143,96 @@ export default {
...
+
+    dislikePlaceReview: (context, data) => {
+        httpService.post(`reviews/${data.id}/dislike`)
+            .then( (res) => {
+                let review = context.state.places.find( (place) => { 
+                    return place.id === parseInt(data.placeId); 
+                }).review;
+                if (review.like === STATUS_NONE) {
+                    context.commit('SET_PLACE_REVIEW_LIKE_STATE', {
+                        placeId: data.placeId,
```
Comment (nemoipaha) [inline frontend/src/store/modules/place/actions.js]: same as above

File: frontend/src/store/modules/place/mutations.js [hunk 4]
```diff
@@ -41,5 +41,23 @@ export default {
 
     SET_CURRENT_PLACE_REVIEW_DISLIKE_COUNT: (state, { reviewId, count }) => {
         state.currentPlaceReviews.byId[reviewId].dislikes = count;
+    },
+
+    SET_PLACE_REVIEW_LIKE_STATE: (state, { placeId, likeState }) => {
+        state.places.find( (place) => {
```
Comment (nemoipaha) [inline frontend/src/store/modules/place/mutations.js]: create findPlaceByReview(reviewId) function and move logic inside
```
const place = state.places.find( (place) => { 
      return place.review.id === parseInt(placeId); 
});

return place;
```
mutation would look like `SET_PLACE_REVIEW_LIKE_STATE`
```
SET_PLACE_REVIEW_LIKE_STATE((state, { reviewId, likeState })) {
   const place = findPlaceByReview(state.places, reviewId);
   place.review.like = likeState;
}
```

File: frontend/src/store/modules/place/mutations.js [hunk 5]
```diff
@@ -41,5 +41,23 @@ export default {
...
+    },
+
+    SET_PLACE_REVIEW_LIKE_STATE: (state, { placeId, likeState }) => {
+        state.places.find( (place) => { 
+            return place.id === parseInt(placeId); 
+        }).review.like = likeState;
+    },
+
+    SET_PLACE_REVIEW_LIKE_COUNT: (state, { placeId, count }) => {
+        state.places.find( (place) => {
```
Comment (nemoipaha) [inline frontend/src/store/modules/place/mutations.js]: same as above

File: frontend/src/store/modules/place/actions.js [hunk 6]
```diff
@@ -143,6 +143,96 @@ export default {
...
+                        likeState: STATUS_LIKED
+                    });
+                    context.commit('SET_PLACE_REVIEW_LIKE_COUNT', {
+                        placeId: data.placeId,
+                        count: review.likes + 1
+                    });
+                } else if (review.like === STATUS_LIKED) {
```
Comment (nemoipaha) [inline frontend/src/store/modules/place/actions.js]: All logic for 
if (review.like === STATUS_NONE) {...} 
else if (if (review.like === STATUS_LIKED) {) {...}
can be own action method, `updateReviewLikeStatus(review, placeId)`

File: frontend/src/store/modules/place/mutations.js [hunk 7]
```diff
@@ -1,3 +1,11 @@
+import {STATUS_LIKED, STATUS_DISLIKED, STATUS_NONE} from '@/services/api/codes';
+
+function findReviewById (state, reviewId) {
```
Comment (nemoipaha) [inline frontend/src/store/modules/place/mutations.js]: better to make it like findReviewById(places, reviewId)

Comment (ivanshamis) [reply frontend/src/store/modules/place/mutations.js]: fixed

**Pull request merged**